engine/src/main/java/org/hibernate/search/engine/search/query/spi/QueryParameters.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.query.spi;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.search.engine.logging.impl.Log;
+import org.hibernate.search.util.common.annotation.Incubating;
+import org.hibernate.search.util.common.impl.Contracts;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+@Incubating
+public class QueryParameters {
```

Comment:
I went with a "dedicated" wrapper around the map for these query parameters so that we can have various parameter checks in one place. See also a couple of TODOs in this class. I think that it doesn't make sense to have a parameter with a blank name or a null value. At least I'm not sure that there's a useful use case when the value is `null` ...